script.js

```diff
@@ -1,4 +1,408 @@
+    // Select a random index in the deck
+    var randomIndex = getRandomIndex(cardDeck.length);
+    // Select the card that corresponds to randomIndex
+    var randomCard = cardDeck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    var currentCard = cardDeck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    cardDeck[currentIndex] = randomCard;
+    cardDeck[randomIndex] = currentCard;
+    // Increment currentIndex
+    currentIndex = currentIndex + 1;
+  }
+  // Return the shuffled deck
+  return cardDeck;
+};
+
+// Shuffle the deck and save it in a new variable shuffledDeck
+// to communicate that we have shuffled the deck.
+var pickCard = function () {
+  card = cardDeck.pop();
+  return card;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  //shuffle the deck with the first submit click
+  if (gameMode == "shuffle") {
+    shuffledDeck = shuffleCards(cardDeck);
+    console.log(shuffledDeck);
+    gameMode = "player pick";
+    return "The cards have been shuffled and the game is ready to begin!";
+  }
+  //initial player pick gets displayed and the count is recorded
+  if (gameMode == "player pick") {
+    gameMode = "player further picks or computer";
+    playerCard1 = pickCard(shuffledDeck);
+    playerCard2 = pickCard(shuffledDeck);
+    //case when player gets an ace
+    if (playerCard1.rank == 1 && playerCard2.rank < 11) {
+      playerCard1.rank = 11;
+    }
+    if (playerCard2.rank == 1 && playerCard1.rank < 11) {
+      playerCard2.rank = 11;
+    }
+    playerCount = playerCard1.rank + playerCard2.rank;
+    console.log(playerCount);
+    return `Your first two cards are ${playerCard1.name} of ${playerCard1.suit} and ${playerCard2.name} of ${playerCard2.suit}. Your total count sits at ${playerCount}<br><br>If you want to pick another card please click 'submit', else please input 'computer' into the input box to let the computer play!`;
+  }
+  //player picks another card
+  if (gameMode == "player further picks or computer" && input == "") {
+    playerCard1 = pickCard(shuffledDeck);
+    //case when player gets an ace
+    if (playerCard1.rank == 1 && playerCount < 11) {
+      playerCard1.rank = 11;
+    }
+    playerCount = playerCount + playerCard1.rank;
+    var resultString = `You have picked ${playerCard1.name} of ${playerCard1.suit} and your total count sits at ${playerCount}<br><br>`;
+    console.log(playerCount);
+
+    if (playerCount > 21) {
+      var bust = true;
+      gameMode = "shuffle";
+      resultString +=
+        "You've gone bust and you lose! Please press 'submit' if you want to play again.<br><br>";
+    }
+
+    if (bust == true) {
+      return resultString;
+    } else {
+      return `${resultString} If you want to pick another card please click 'submit', else please input 'computer' into the input box to let the computer play!`;
+    }
+  }
```

Comment:
This code could have been combine.
 Example 

 if (playerCount > 21) {
      gameMode = "shuffle";
      resultString +=
        "You've gone bust and you lose! Please press 'submit' if you want to play again";
    } else {
      return `${resultString} If you want to pick another card please click 'submit', else please input 'computer' into the input box to let the computer play!`;
    }
    
    therefore making "if (bust == true)" redundant .